File: ZPRO/prednasky/prednaska7/list.cpp
#include "list.hpp"

void CreateList(list& s){
    s.head = s.sentinel = new element {NULA, nullptr };
    s.length = 0;
}

void Push_front(list& s, T what){
    s.head = new element{what, s.head};
    s.length ++;
}

void Push_back(list& s, T what){
    s.sentinel-> data = what;
    s.sentinel -> next = new element{ what, nullptr};
    s.sentinel = s.sentinel -> next;
    s.length ++;
}

void Pop_first(list& s ){
    auto tmp = s.head;
    s.head = s.head -> next;
    delete tmp;
    s.length --;
}

File: PJ/Lecture2/listimplementation/List.java
package PJ.Lecture2.listimplementation;

public class List {
    public List ()
    {
        head = tail = new Element();
    }
    public void pushFront (Object data)
    {
        head = new Element(data, head);
    }
    public boolean empty()
    {
        return head == tail;
    }
    public void print ()
    {
        if(empty())
        {
            System.out.println("Empty list :(");
            return;
        }
        Element tmp = head;
        while (tmp !=tail)
        {
            System.out.print(tmp.getData() + ", ");
            tmp = tmp.getNext();
        }
    }
    private Element head;
    private Element tail;
}
